File: test/fixtures/objcpush/android/src/main/java/com/example/objcpush/ObjcpushPlugin.java
package com.example.objcpush;

import com.getcapacitor.JSObject;
import com.getcapacitor.Plugin;
import com.getcapacitor.PluginCall;
import com.getcapacitor.PluginMethod;
import com.getcapacitor.annotation.CapacitorPlugin;

@CapacitorPlugin(name = "Objcpush")
public class ObjcpushPlugin extends Plugin {

    private int pingCount = 0;

    @PluginMethod
    public void echo(PluginCall call) {
        String message = call.getString("message", "hello");
        pingCount = pingCount + 1;
        JSObject response = new JSObject();
        response.put("message", message);
        response.put("count", pingCount);
        call.resolve(response);
    }

    @PluginMethod
    public void ping(PluginCall call) {
        if (pingCount <= 0) {
            call.reject("ping unavailable before echo");
            return;
        }
        notifyListeners("pushReceived", new JSObject() {{
            put("token", "fixed-token");
        }});
        call.resolve();
    }
}

File: test/fixtures/objcpush/ios/Sources/ObjcpushPlugin/ObjcpushPlugin.m
// The FIRST ObjC-native fixture — establishes the single-file convention:
// CAP_PLUGIN registration macro + @implementation in one .m (the @capacitor/
// keyboard two-file split — registration .m + body .m — is also supported).
#import <Foundation/Foundation.h>
#import <Capacitor/Capacitor.h>

CAP_PLUGIN(ObjcpushPlugin, "Objcpush",
           CAP_PLUGIN_METHOD(echo, CAPPluginReturnPromise);
           CAP_PLUGIN_METHOD(ping, CAPPluginReturnPromise);
           CAP_PLUGIN_METHOD(resizeProbe, CAPPluginReturnPromise);
           CAP_PLUGIN_METHOD(windowPing, CAPPluginReturnPromise);
)

@interface ObjcpushPlugin ()
@property (readwrite, assign, nonatomic) NSInteger pingCount;
@end

@implementation ObjcpushPlugin

- (void)echo:(CAPPluginCall *)call
{
  NSString *message = [call getString:@"message" defaultValue:@"hello"];
  self.pingCount = self.pingCount + 1;
  NSDictionary *response = @{@"message": message, @"count": [NSNumber numberWithInteger:self.pingCount]};
  [call resolve:response];
}

- (void)ping:(CAPPluginCall *)call
{
  if (self.pingCount <= 0) {
    [call reject:@"ping unavailable before echo"];
    return;
  }
  NSDictionary *payload = @{@"token": @"fixed-token"};
  [self notifyListeners:@"pushReceived" data:payload];
  [call resolve];
}

// webview-dependent body: frame math against self.webView + a self.bridge JS
// eval — under the no-webview scope policy this MUST become a
// DROP-WITH-REJECTION stub (selector kept, immediate fulfillWithError)
- (void)resizeProbe:(CAPPluginCall *)call
{
  NSString *spec = [call getString:@"payload" defaultValue:@"{}"];
  CGRect frame = self.webView.frame;
  [self.bridge evalWithJs:[NSString stringWithFormat:@"resize(%f)", frame.size.height]];
  [call resolve];
}

// bridge window-JS event with a CONSTANT name that this method does NOT also
// notifyListeners — MUST reroute through the locked-listener emit
- (void)windowPing:(CAPPluginCall *)call
{
  [self.bridge triggerWindowJSEventWithEventName:@"windowPinged" data:@"from-window"];
  [call resolve];
}

@end
